src/lib/y2storage/blk_device.rb

```diff
@@ -315,5 +328,15 @@ def basename
     def hwinfo
       Y2Storage::HWInfoReader.instance.for_device(name)
     end
+
+    # Checks whether the device is in use
+    #
+    # @note A device is in use when it is used as physical volume, belongs to a
+    #   multipath or belongs to a raid.
+    #
+    # @return [Boolean]
+    def used?
```

Comment:
I don't like the name of the method. A disk with formatted partitions is also "used" from some point of view.

Even worst, this method lives in `BlkDevice` but it internally calls `#partittion_table` which is defined in `Partitionable`. I would expect fireworks if this method is called on a `Encryption` or `LvmLv`, that are also block devices. And I would expect it to always return false for objects of type `Partition` (`#partition_table` cannot be nil) 

So this method should be in `Partitionable`. About the name... that's tricky. It checks whether the partitionable as a whole (not through its partitions) is used as a component of some other bigger device (like a RAID, a VG or a multipath).

Reply:
I would even give up on providing such a specific check directly in `Y2Storage::Partitionable`. It's a one-liner, so it can be checked directly in the controller of the action in the partitioner. Specially taking into account that BIOS RAIDs and Multipath are kind of irrelevant there, since those disks will not be shown anyways.